File: backend/models/product.py
from sqlalchemy import Column, Integer, String, Float, ForeignKey
from sqlalchemy.orm import relationship
from database import Base


class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    name = Column(String, nullable=False)
    description = Column(String, nullable=True)
    price = Column(Float, nullable=False)
    stock = Column(Integer, nullable=False, default=0)
    category = Column(String, nullable=True)
    image_url = Column(String, nullable=True)

    #  link product to the vendor (User) who owns it
    vendor_id = Column(Integer, ForeignKey("users.id"), nullable=True)

    # Optional but useful: lets you do product.vendor.name in Python
    vendor = relationship("User", foreign_keys=[vendor_id])

File: frontend/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String? description;
  final double price;
  final int stock;
  final String? category;
  final String? imageUrl;
  final int? vendorId;
  final String? vendorName;

  Product({
    required this.id, required this.name,
    this.description,
    required this.price, required this.stock,
    this.category, this.imageUrl,
    this.vendorId, this.vendorName,
  });

  factory Product.fromJson(Map<String, dynamic> json) => Product(
    id: json['id'],
    name: json['name'],
    description: json['description'],
    price: (json['price'] as num).toDouble(),
    stock: json['stock'],
    category: json['category'],
    imageUrl: json['image_url'],
    vendorId: json['vendor_id'],
    vendorName: json['vendor_name'],
  );

  Map<String, dynamic> toJson() => {
    'name': name,
    'description': description,
    'price': price,
    'stock': stock,
    'category': category,
    'image_url': imageUrl,
    'vendor_id': vendorId,
  };
}